### PR TITLE
Add 1.4.0 to readme compatibility matrix 

### PR DESCRIPTION
**Pull request by xinyanw409**
**What this PR does / why we need it**:
This PR added 1.4.0 to readme compatibility matrix.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add 1.4.0 to readme compatibility matrix.
```

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/746/

File: docs/guest.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ This document discusses the velero vSphere plugin installation process in a **Ta
 
 | Velero Plugin for vSphere Version | vSphere Version    | Kubernetes Version | vSphere CSI Driver Version | Velero Version | Deprecated | EOL Date      |
 |-----------------------------------|--------------------|--------------------|----------------------------|----------------|------------|---------------|
+| 1.4.0                             | 7.0U1c/P02 - 7.0U3 | 1.19-1.21          | Bundled with TKGS          | 1.8.1          | No         | N/A
```
Comment (xing-yang) [inline docs/guest.md:19]: I thought 1.22 is tested.

File: docs/supervisor.md [hunk 2]
```diff
@@ -11,14 +11,15 @@
...
-| 1.2.0                             | 7.0U1c/P02 - 7.0U2 | Bundled with vSphere (1.16-1.19, 1.18-1.20)            | Bundled with vSphere       | 1.5.1          | Yes        | December 2022 |
-| 1.1.1                             | 7.0U1c/P02         | Bundled with vSphere (1.16-1.19)                       | Bundled with vSphere       | 1.5.1          | No         | N/A           |
-| 1.1.0                             | 7.0U1c/P02         | Bundled with vSphere (1.16-1.19)                       | Bundled with vSphere       | 1.5.1          | Yes        | December 2022 |
+| Velero Plugin for vSphere Version | vSphere Version    | Kubernetes Version                                                | vSphere CSI Driver Version | Velero Version | Deprecated | EOL Date      |
+|-----------------------------------|--------------------|-------------------------------------------------------------------|----------------------------|----------------|------------|---------------|
+| 1.4.0                             | 7.0U1c/P02 - 7.0U3 | Bundled with vSphere (1.16-1.19, 1.18-1.20, 1.19-1.21)            | Bundled with vSphere       | 1.8.1          | No         | N/A
```
Comment (xing-yang) [inline docs/supervisor.md:16]: K8s 1.22 should be tested.

File: docs/vanilla.md [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 | Velero Plugin for vSphere Version | vSphere Version        | Kubernetes Version | vSphere CSI Driver Version | Velero Version | Deprecated | EOL Date      |
 |-----------------------------------|------------------------|--------------------|----------------------------|----------------|------------|---------------|
+| 1.4.0                             | 6.7U3 P06 - 7.0U3      | 1.20-1.23          | 2.2.2, 2.3.1, 2.4.1, 2.5.1 | 1.8.1          | No         | N/A
```
Comment (rpanduranga) [inline docs/vanilla.md:17]: CSI 2.6 is also supported and validated

File: docs/supervisor.md (hunk 2 above)
Comment (rpanduranga) [reply docs/supervisor.md:16]: K8s 1.22 is tested

File: docs/guest.md (hunk 1 above)
Comment (rpanduranga) [reply docs/guest.md:19]: k8s 1.22 is tested

**Review by rpanduranga — CHANGES_REQUESTED** [review]

**Review by rpanduranga — APPROVED** [review]

**Review by deepakkinni — APPROVED** [review]

**Pull request merged**